robottelo/cli/factory.py

```diff
@@ -3635,3 +3715,377 @@ def add_role_permissions(role_id, resource_permissions):
+                'url': custom_repo_url,
+            })
+        if download_policy:
+            # Set download policy
+            Repository.update({
+                'download-policy': download_policy,
+                'id': repo_info['id'],
+            })
+        repos_info.append(repo_info)
+    # Synchronize the repositories
+    for repo_info in repos_info:
+        Repository.synchronize({'id': repo_info['id']})
+    # Create a content view
+    content_view_id = make_content_view({u'organization-id': org_id})['id']
+    # Add repositories to content view
+    for repo_info in repos_info:
+        ContentView.add_repository({
+            u'id': content_view_id,
+            u'organization-id': org_id,
+            u'repository-id': repo_info['id'],
+        })
+    # Publish the content view
+    ContentView.publish({u'id': content_view_id})
+    # Get the latest content view version id
+    content_view_version = ContentView.info({
+            u'id': content_view_id
+        })['versions'][-1]
+    # Promote content view version to lifecycle environment
+    ContentView.version_promote({
+        u'id': content_view_version['id'],
+        u'organization-id': org_id,
+        u'to-lifecycle-environment-id': lce_id,
+    })
+    content_view = ContentView.info({u'id': content_view_id})
+    activation_key = make_activation_key({
+        u'organization-id': org_id,
+        u'lifecycle-environment-id': lce_id,
+        u'content-view-id': content_view_id,
+    })
+    # Get organization subscriptions
+    subscriptions = Subscription.list({
+        u'organization-id': org_id},
+        per_page=False
+    )
+    # Add subscriptions to activation-key
+    needed_subscription_names = list(rh_subscriptions)
+    if custom_product:
+        needed_subscription_names.append(custom_product['name'])
+    added_subscription_names = []
+    for subscription in subscriptions:
+        if subscription['name'] in needed_subscription_names:
+            ActivationKey.add_subscription({
+                u'id': activation_key['id'],
+                u'subscription-id': subscription['id'],
+                u'quantity': 1,
+            })
+            added_subscription_names.append(subscription['name'])
+            if (len(added_subscription_names)
+                    == len(needed_subscription_names)):
+                break
+    missing_subscription_names = set(
+        needed_subscription_names).difference(set(added_subscription_names))
+    if missing_subscription_names:
+        raise CLIFactoryError(
+            u'Missing subscriptions: {0}'.format(missing_subscription_names))
+
+    return dict(
+        activation_key=activation_key,
+        content_view=content_view,
+        repos=repos_info,
+    )
+
+
+def setup_vm_ssh_key_host_access(
+        vm, local_key_path=None, remote_key_name=None, remote_ssh_path=None,
+        access_hostname=None, access_user=None):
+    """Configure a Virtual machine to access a host using ssh key
```

Comment:
The implementation is ok, but var names and docstring are very confusing as for me (or i got them wrong).
`Configure a Virtual machine to access a host using ssh key` - what host? `access_hostname`? But it's optional, function can be executed without specifying it. From what i've got, it's rather smth like 'upload ssh key to vm and (optionally) add desired host to known_hosts'. If that's the case, then we should consider dividing this func to 2 separate ones - what if we uploaded ssh key already and wanna add few known_hosts?
Second, `local_key_path` and `remote_ssh_path` are misleading here. 'local' here is used to specify satellite's key and 'remote path' - vm's path. But it's vm's helper, from vm perspective naming should be opposite - the path is local and the key is remote 🙂 So i'd suggest to use different approach for naming here - let's say `sat_key_path` and `vm_ssh_path` to avoid confusion.